internal/clients/nightfall/nightfall_api.go

```diff
@@ -8,17 +8,23 @@ import (
 // APIClient is a wrapper around nightfallAPI.APIClient
 type APIClient struct {
 	APIClient *nightfallAPI.APIClient
+	ScanApi   nightfallintf.NightfallScanAPI
 }
 
 // NewAPIClient generates a nightfallAPI client
-func NewAPIClient() *APIClient {
+func NewAPIClient(scanAPI nightfallintf.NightfallScanAPI) *APIClient {
```

Comment:
this all seems a bit weird to me. Are we sure we need this `ScanAPI` field both behind APIClient and adjacent to it?

Can we just move the `ScanPayload` method so that `APIClient` is its receiver?

Haven't played much w/ it or done much testing but I mean something like this: https://github.com/nightfallai/nightfall_cli/pull/26/files